core/src/main/java/org/elasticsearch/action/count/CountRequest.java

```diff
@@ -254,7 +199,16 @@ public SearchRequest toSearchRequest() {
         if (terminateAfter() != DEFAULT_TERMINATE_AFTER) {
             searchSourceBuilder.terminateAfter(terminateAfter());
```

Comment:
terminateAfter and minScore were treated differently because QuerySourceBuilder didn't support them before. Shall we now just remove their setters as they are part of SearchSourceBuilder? Then we would get rid of this unnecessary handover of values.